### PR TITLE
Add @medplum/health-gorilla-react package

### PR DESCRIPTION
**Pull request by mattlong**
Part 2 of #5205

### CODE REVIEW

File: packages/health-gorilla-react/package.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    "jsdom": "25.0.0",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-router-dom": "6.26.2",
+    "rimraf": "5.0.9",
+    "vitest": "2.0.5"
+  },
+  "peerDependencies": {
+    "@medplum/core": "3.2.10",
+    "@medplum/fhirtypes": "3.2.10",
```
Comment (ThatOneBro) [inline packages/health-gorilla-react/package.json:82]: I think we need a `@medplum/health-gorilla-core` here

File: packages/health-gorilla-react/src/autocomplete-endpoint.test.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        let result: Questionnaire | undefined;
+        if (invalidResultType) {
+          result = { resourceType: 'Patient', name: [{ text: 'Test Patient' }] } as unknown as Questionnaire;
+        } else {
+          result = input.testCode.code ? ALL_AOES[input.testCode.code] : undefined;
+        }
+        return { outcome: allOk, type, result };
+      }
+      default: {
+        input satisfies never;
```
Comment (ThatOneBro) [inline packages/health-gorilla-react/src/autocomplete-endpoint.test.ts:168]: Interesting, very smart

File: packages/health-gorilla-react/src/useHealthGorillaLabOrder.test.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        <MemoryRouter>
+          <MedplumProvider medplum={medplum}>{children}</MedplumProvider>
+        </MemoryRouter>
+      ),
+    });
+  }
+
+  function setupContext(
+    hookOptions: UseHealthGorillaLabOrderOptions,
+    withoutProvider?: boolean
```
Comment (ThatOneBro) [inline packages/health-gorilla-react/src/useHealthGorillaLabOrder.test.tsx:101]: Neat

File: packages/health-gorilla-core/src/lab-order.ts [hunk 4]
```diff
@@ -345,7 +345,7 @@ export function createLabOrderBundle(inputs: PartialLabOrderInputs): Bundle {
   return bundle;
 }
 
-function formatTestCoding(test: TestCoding): string {
+export function formatTestCoding(test: TestCoding): string {
```
Comment (codyebberson) [inline packages/health-gorilla-core/src/lab-order.ts:348]: Should we merge this into `formatCoding` in `packages\core\src\format.ts`?

File: packages/health-gorilla-core/src/test-utils.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { expect } from 'vitest';
+
+export function expectToBeDefined<T>(value: T | undefined, message?: string): asserts value is T {
```
Comment (codyebberson) [inline packages/health-gorilla-core/src/test-utils.ts:3]: extreme nitpick, but to an outside observer, this seems like more noise than utility

```
expect(x, msg).toBeDefined();
expectToBeDefined(x, msg);
```

File: packages/health-gorilla-react/package.json [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    "react-hooks"
+  ],
+  "homepage": "https://www.medplum.com/",
+  "bugs": {
+    "url": "https://github.com/medplum/medplum/issues"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/medplum/medplum.git",
+    "directory": "packages/core"
```
Comment (codyebberson) [inline packages/health-gorilla-react/package.json:29]:
```suggestion
    "directory": "packages/health-gorilla-react"
```

File: packages/health-gorilla-react/src/test.setup.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { afterEach } from 'vitest';
+import { cleanup } from '@testing-library/react';
+// import * as matchers from '@testing-library/jest-dom/matchers';
```
Comment (codyebberson) [inline packages/health-gorilla-react/src/test.setup.ts:3]: Cleanup

```suggestion
```

**Review by codyebberson — APPROVED** [review]
Thanks for doing this 🙏 

Minor nits, but nothing major 👍

File: packages/health-gorilla-core/src/test-utils.ts (hunk 5 above)
Comment (mattlong) [reply packages/health-gorilla-core/src/test-utils.ts:3]: Yeah, very true. The difference is the type assertion which makes following lines of test code more precise.

**Pull request merged**